### DISCUSSION TITLE
Entry point of sea horn

### DISCUSSION THREAD

**Discussion by Nivous** [seahorn/seahorn#516] [General]
Hi. I am looking to add functionality to the sea horn tool, to support more complicated invariants.
I could not figure out what is the entry point of sea horn (i.e what is the first function being called from the python code that parses the arguments).

**Comment by agurfinkel** [comment]
The python script `sea` schedules multiple executables to be called for different pre-processing passes. The last pass is `seahorn` binary that is responsible for actual analysis. The entry point for `seahorn` binary is here: https://github.com/seahorn/seahorn/blob/main/tools/seahorn/seahorn.cpp#L217
